### ISSUE TITLE
Publish gem after was once revoked, and publishes, but returns error

### ISSUE THREAD

**Issue by justin808** [svenfuchs/gem-release#63]
Gem is published. Server still sends back an error mesage.

This is for a very active gem, https://github.com/shakacode/react_on_rails

Using this repo: https://github.com/svenfuchs/gem-release

and this script:

https://github.com/shakacode/react_on_rails/blob/master/rakelib/release.rake#L64

```
+ react-on-rails@6.8.2
No distRepo provided, done.
cd /Users/justin/shakacode/react-on-rails/react_on_rails && gem release
WARNING:  open-ended dependency on connection_pool (>= 0) is not recommended
  if connection_pool is semantically versioned, use:
    add_runtime_dependency 'connection_pool', '~> 0'
WARNING:  open-ended dependency on rails (>= 3.2) is not recommended
  if rails is semantically versioned, use:
    add_runtime_dependency 'rails', '~> 3.2'
WARNING:  open-ended dependency on addressable (>= 0) is not recommended
  if addressable is semantically versioned, use:
    add_runtime_dependency 'addressable', '~> 0'
WARNING:  open-ended dependency on rspec (>= 0, development) is not recommended
  if rspec is semantically versioned, use:
    add_development_dependency 'rspec', '~> 0'
WARNING:  open-ended dependency on coveralls (>= 0, development) is not recommended
  if coveralls is semantically versioned, use:
    add_development_dependency 'coveralls', '~> 0'
WARNING:  open-ended dependency on generator_spec (>= 0, development) is not recommended
  if generator_spec is semantically versioned, use:
    add_development_dependency 'generator_spec', '~> 0'
WARNING:  open-ended dependency on pry (>= 0, development) is not recommended
  if pry is semantically versioned, use:
    add_development_dependency 'pry', '~> 0'
WARNING:  open-ended dependency on pry-byebug (>= 0, development) is not recommended
  if pry-byebug is semantically versioned, use:
    add_development_dependency 'pry-byebug', '~> 0'
WARNING:  open-ended dependency on pry-stack_explorer (>= 0, development) is not recommended
  if pry-stack_explorer is semantically versioned, use:
    add_development_dependency 'pry-stack_explorer', '~> 0'
WARNING:  open-ended dependency on pry-doc (>= 0, development) is not recommended
  if pry-doc is semantically versioned, use:
    add_development_dependency 'pry-doc', '~> 0'
WARNING:  open-ended dependency on pry-state (>= 0, development) is not recommended
  if pry-state is semantically versioned, use:
    add_development_dependency 'pry-state', '~> 0'
WARNING:  open-ended dependency on pry-toys (>= 0, development) is not recommended
  if pry-toys is semantically versioned, use:
    add_development_dependency 'pry-toys', '~> 0'
WARNING:  open-ended dependency on pry-rescue (>= 0, development) is not recommended
  if pry-rescue is semantically versioned, use:
    add_development_dependency 'pry-rescue', '~> 0'
WARNING:  open-ended dependency on binding_of_caller (>= 0, development) is not recommended
  if binding_of_caller is semantically versioned, use:
    add_development_dependency 'binding_of_caller', '~> 0'
WARNING:  open-ended dependency on awesome_print (>= 0, development) is not recommended
  if awesome_print is semantically versioned, use:
    add_development_dependency 'awesome_print', '~> 0'
WARNING:  open-ended dependency on gem-release (>= 0, development) is not recommended
  if gem-release is semantically versioned, use:
    add_development_dependency 'gem-release', '~> 0'
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: react_on_rails
  Version: 6.8.2
  File: react_on_rails-6.8.2.gem
Pushing gem to https://rubygems.org...
Successfully registered gem: react_on_rails (6.8.2)
Deleting left over gem file react_on_rails-6.8.2.gem
WARNING:  open-ended dependency on connection_pool (>= 0) is not recommended
  if connection_pool is semantically versioned, use:
    add_runtime_dependency 'connection_pool', '~> 0'
WARNING:  open-ended dependency on rails (>= 3.2) is not recommended
  if rails is semantically versioned, use:
    add_runtime_dependency 'rails', '~> 3.2'
WARNING:  open-ended dependency on addressable (>= 0) is not recommended
  if addressable is semantically versioned, use:
    add_runtime_dependency 'addressable', '~> 0'
WARNING:  open-ended dependency on rspec (>= 0, development) is not recommended
  if rspec is semantically versioned, use:
    add_development_dependency 'rspec', '~> 0'
WARNING:  open-ended dependency on coveralls (>= 0, development) is not recommended
  if coveralls is semantically versioned, use:
    add_development_dependency 'coveralls', '~> 0'
WARNING:  open-ended dependency on generator_spec (>= 0, development) is not recommended
  if generator_spec is semantically versioned, use:
    add_development_dependency 'generator_spec', '~> 0'
WARNING:  open-ended dependency on pry (>= 0, development) is not recommended
  if pry is semantically versioned, use:
    add_development_dependency 'pry', '~> 0'
WARNING:  open-ended dependency on pry-byebug (>= 0, development) is not recommended
  if pry-byebug is semantically versioned, use:
    add_development_dependency 'pry-byebug', '~> 0'
WARNING:  open-ended dependency on pry-stack_explorer (>= 0, development) is not recommended
  if pry-stack_explorer is semantically versioned, use:
    add_development_dependency 'pry-stack_explorer', '~> 0'
WARNING:  open-ended dependency on pry-doc (>= 0, development) is not recommended
  if pry-doc is semantically versioned, use:
    add_development_dependency 'pry-doc', '~> 0'
WARNING:  open-ended dependency on pry-state (>= 0, development) is not recommended
  if pry-state is semantically versioned, use:
    add_development_dependency 'pry-state', '~> 0'
WARNING:  open-ended dependency on pry-toys (>= 0, development) is not recommended
  if pry-toys is semantically versioned, use:
    add_development_dependency 'pry-toys', '~> 0'
WARNING:  open-ended dependency on pry-rescue (>= 0, development) is not recommended
  if pry-rescue is semantically versioned, use:
    add_development_dependency 'pry-rescue', '~> 0'
WARNING:  open-ended dependency on binding_of_caller (>= 0, development) is not recommended
  if binding_of_caller is semantically versioned, use:
    add_development_dependency 'binding_of_caller', '~> 0'
WARNING:  open-ended dependency on awesome_print (>= 0, development) is not recommended
  if awesome_print is semantically versioned, use:
    add_development_dependency 'awesome_print', '~> 0'
WARNING:  open-ended dependency on gem-release (>= 0, development) is not recommended
  if gem-release is semantically versioned, use:
    add_development_dependency 'gem-release', '~> 0'
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: react_on_rails
  Version: 6.8.2
  File: react_on_rails-6.8.2.gem
Pushing gem to https://rubygems.org...
Repushing of gem versions is not allowed.
Please use `gem yank` to remove bad gem releases.
rake aborted!
Command failed with status (1): [cd /Users/justin/shakacode/react-on-rails/...]
/Users/justin/shakacode/react-on-rails/react_on_rails/rakelib/task_helpers.rb:20:in `block in sh_in_dir'
/Users/justin/shakacode/react-on-rails/react_on_rails/rakelib/task_helpers.rb:20:in `each'
/Users/justin/shakacode/react-on-rails/react_on_rails/rakelib/task_helpers.rb:20:in `sh_in_dir'
/Users/justin/shakacode/react-on-rails/react_on_rails/rakelib/release.rake:64:in `block in <top (required)>'
/Users/justin/.rvm/gems/ruby-2.3.1@react_on_rails/gems/rake-12.0.0/exe/rake:27:in `<top (required)>'
Tasks: TOP => release
(See full trace by running task with --trace)
```

See [#1589](https://github.com/rubygems/rubygems.org/issues/1589) and [Issue on RubyGems](http://help.rubygems.org/discussions/problems/25897-after-i-yanked-a-version-i-can-never-push-without-an-error).

**Comment by dwradcliffe** [comment]
See also https://github.com/rubygems/rubygems.org/issues/1577

**Comment by isolo** [comment]
In that case, the problem is that release command uses method [in_gemspec_dirs](https://github.com/svenfuchs/gem-release/blob/master/lib/gem_release/helpers.rb#L53) which yielded all gem specs it can find. React_on_rails has examples/specs that contains react_on_rails gemspec. Subsequently, the release command is invoked many times.

**Comment by justin808** [comment]
I found the issue:

```
find . -name "*.gemspec"
./react_on_rails.gemspec
./spec/dummy/client/node_modules/react-on-rails/react_on_rails.gemspec
```

That comes from: https://github.com/shakacode/react_on_rails/blob/master/spec/dummy/client/package.json#L36

```
    "react-on-rails": "file:../../..",
```


@isolo Any chance that you can take a look at the rake task for the release and delete that extra gemspec? Or maybe delete all gemspecs except the root one? I think the properly crafted find/rm command right here could do the trick.

https://github.com/shakacode/react_on_rails/blob/master/rakelib/release.rake#L29

**Comment by justin808** [comment]
Moving discussion to React on Rails.

**Comment by svenfuchs** [comment]
@justin808 I believe your issue should be fixed in `v2.0.0.pre.x` which gives you more control over this. See https://github.com/svenfuchs/gem-release/pull/64 and the [readme](https://github.com/svenfuchs/gem-release/blob/sf-v2/README.md).

As far as I understand [this scenario](https://github.com/svenfuchs/gem-release/tree/sf-v2#scenario-4-nested-gem-with-a-conventional-sub-directory-name) is similar to your case. As you can see `gem bump` would only bump the gem in the top-level directory in this case. `gem release` uses the same logic for detecting/selecting gems to operate on.

If that does not work, then you can specify `gem release react_on_rails` in order to make sure only this one gem is used.

I am going to close this here. Please reopen if it doesn't work as expected.
